pkg/daemon/config.go

```diff
@@ -92,3 +106,55 @@ func tryToLoadOldConfig(nodeProfilesJson []byte) ([]ptpv1.PtpProfile, bool) {
 
 	return []ptpv1.PtpProfile{*ptpConfig}, true
 }
+
+func (conf *ptp4lConf) populatePtp4lConf(config *string) error {
+	lines := strings.Split(*config, "\n")
+	var currentSection string
+	conf.sections = make(map[string]ptp4lConfSection)
+
+	for _, line := range lines {
+		if strings.HasPrefix(line, "[") {
+			currentSection = line
+			currentLine := strings.Split(line, "]")
+
+			if len(currentLine) < 1 {
+				return errors.New("Section missing closing ']'")
+			}
+
+			currentSection = fmt.Sprintf("%s]", currentLine[0])
+			section := ptp4lConfSection{}
+			section.options = make(map[string]string)
+			conf.sections[currentSection] = section
+			if currentSection != "[global]" {
+				iface := currentSection
+				iface = strings.ReplaceAll(iface, "[", "")
+				iface = strings.ReplaceAll(iface, "]", "")
+				conf.mapping = append(conf.mapping, iface)
+				conf.ifaceString = fmt.Sprintf("%s%s", conf.ifaceString, line)
+			}
+		} else if currentSection != "" {
+			split := strings.IndexByte(line, ' ')
+			if split > 0 {
+				section := conf.sections[currentSection]
+				section.options[line[:split]] = line[split:]
+				conf.sections[currentSection] = section
+			}
+		} else {
+			return errors.New("Config option not in section")
+		}
+	}
+	conf.ifaceString = strings.Join(conf.mapping, ",")
+	return nil
+}
+
+func (conf *ptp4lConf) renderPtp4lConf() {
+
+	conf.conf = ""
```

Comment:
no need for conf we are not using it at all.
this function just need to return a string and that's it